src/cnd/swaps_payload.ts

```diff
@@ -0,0 +1,312 @@
+  Herc20Request &
+  EthereumRequest;
+
+export type HanEthereumEtherHalightLightningBitcoinRequestBody = CoreRequestBody<
+  HanEthereumEtherRequestParams,
+  HalightLightningBitcoinRequestParams
+>;
+export type Herc20EthereumErc20HalightLightningBitcoinRequestBody = CoreRequestBody<
+  Herc20EthereumErc20RequestParams,
+  HalightLightningBitcoinRequestParams
+>;
+export type HalightLightningBitcoinHanEthereumEtherRequestBody = CoreRequestBody<
+  HalightLightningBitcoinRequestParams,
+  HanEthereumEtherRequestParams
+>;
+export type HalightLightningBitcoinHerc20EthereumErc20RequestBody = CoreRequestBody<
+  HalightLightningBitcoinRequestParams,
+  Herc20EthereumErc20RequestParams
+>;
+
+/**
+ * The payload returned when fetching one swap on the `/swaps/:id` endpoint
+ */
+export interface SwapResponse extends Entity {
+  properties: Properties;
+  entities: Array<LedgerState | LedgerParameters>;
+  actions: SwapAction[];
+  /**
+   * links for this swap, contains a self reference
+   */
+  links: Link[];
+}
+
+/**
+ * Element of the array in the payload returned when fetching all swaps on the `/swaps/` endpoint
+ */
+export interface SwapElementResponse extends EmbeddedRepresentationSubEntity {
+  properties: Properties;
+  entities: Array<LedgerState | LedgerParameters>;
+  actions: SwapAction[];
+  /**
+   * links for this swap, contains a self reference
+   */
+  links: Link[];
+}
+
+/**
+ * The properties of a swap
+ */
+export interface Properties {
+  // id: string; // No id! The id should not be manipulated by the client, navigation should be done using the `links` property.
+  /**
+   * The status this swap is currently in.
+   */
+  status: SwapStatus;
+  /**
+   * The role in which you are participating in this swap.
+   */
+  role: "Alice" | "Bob";
+}
+
+/**
+ * The overall status of a swap
+ */
+export enum SwapStatus {
+  /**
+   * The swap was created but the communication phase hasn't yet been finalized
+   */
+  Created = "CREATED",
+  /**
+   * The communication was finalized and blockchain actions are needed or happening
+   */
+  InProgress = "IN_PROGRESS",
+  /**
+   * The swap is finished and the assets were swapped
+   */
+  Swapped = "SWAPPED",
+  /**
+   * The swap is finished and the assets were not swapped
+   */
+  NotSwapped = "NOT_SWAPPED",
+  /**
+   * An unexpected internal failure aborted the swap
+   */
+  InternalFailure = "INTERNAL_FAILURE"
+}
+
+/**
+ * The parameters of a given ledger
+ */
+export interface LedgerParameters extends EmbeddedRepresentationSubEntity {
+  /**
+   * The relation of these ledger parameters to the parent object (*SwapProperties).
+   */
+  rel: ["alpha" | "beta"];
+  /**
+   * Human readable title.
+   */
+  title: "Parameters of the Alpha Ledger" | "Parameters of the Beta Ledger";
+  /**
+   * Class of this sub-entity to facilitate parsing.
+   */
+  class: ["parameters"];
+  properties: HanEthereum | HanBitcoin | Herc20 | HalightBitcoin;
```

Comment:
@yosriady What do you think about that? Would it be a nightmare to handle?

Reply:
I think this is fine.